File: applicatie/src/main/scala/com/vdzon/administratie/model/Afschrift.scala
package com.vdzon.administratie.model

import java.time.LocalDate

import org.mongodb.morphia.annotations.{Entity, Id}

import scala.annotation.meta.field

@Entity("afschrift") case class Afschrift(
                                           @(Id @field) val uuid: String = null,
                                           val nummer: String = null,
                                           val rekening: String = null,
                                           val omschrijving: String = null,
                                           val relatienaam: String = null,
                                           val boekdatum: LocalDate = null,
                                           val bedrag: Double = .0) {

  def this() = this(uuid = null)

}

File: common/src/main/java/com/vdzon/administratie/model/Afschrift.kt
package com.vdzon.administratie.model


import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import org.litote.kmongo.MongoId

import java.math.BigDecimal
import java.time.LocalDate

class Afschrift(
        @MongoId
        val uuid: String = "",
        val nummer: String = "",
        val rekening: String? = "",
        val omschrijving: String = "",
        val relatienaam: String = "",
        @JsonSerialize(using = LocalDateSerializer::class)
        @JsonDeserialize(using = LocalDateDeserializer::class)
        val boekdatum: LocalDate = LocalDate.now(),
        val bedrag: BigDecimal = BigDecimal.ZERO) {
}
